File: src/rvgen/riscv/zifencei.py
# Copyright 2026 Flavien Solt, Quentin Bordier ETH Zurich.
# Licensed under the General Public License, Version 3.0, see LICENSE for details.
# SPDX-License-Identifier: GPL-3.0-only


from .rvprotoinstrs import *

ZIFENCEI_OPCODE_FENCEI = 0b0001111


# @return uint32_t
def zifencei_fencei(imm: int):
    return instruc_itype(ZIFENCEI_OPCODE_FENCEI, 0b0, 0b001, 0b0, imm)

File: rvgen/src/riscv/zifencei.rs
// Copyright 2026 Flavien Solt, Quentin Bordier ETH Zurich.
// SPDX-License-Identifier: GPL-3.0-only

//! Port of `riscv/zifencei.py`; function names and argument order are retained.

use super::rvprotoinstrs::*;

pub const ZIFENCEI_OPCODE_FENCEI: i32 = 15;

pub fn zifencei_fencei(imm: i32) -> u32 {
    instruc_itype(ZIFENCEI_OPCODE_FENCEI, 0, 1, 0, imm)
}
